### PR TITLE
Allow return rawjson when featuregate is enabled

### PR DESCRIPTION
**Pull request by qiujian16**
relates to https://github.com/open-cluster-management-io/work/issues/196

the feature is enabled when setting RawFeedbackJsonString=true in work agent's feature gate

### CODE REVIEW

File: pkg/spoke/statusfeedback/reader.go [hunk 1]
```diff
@@ -121,6 +127,24 @@ func getValueByJsonPath(name, path string, obj *unstructured.Unstructured) (*wor
...
 		}, nil
+	default:
+		if features.DefaultSpokeMutableFeatureGate.Enabled(ocmfeature.RawFeedbackJsonString) {
+			jsonRaw, err := json.Marshal(&t)
+			if err != nil {
+				return nil, fmt.Errorf("failed to parse the resource to json string: %v", err)
+			}
+			if len(jsonRaw) > maxJsonRawLength {
+				return nil, fmt.Errorf("the length of returned json raw string is larger than the maximum length %d", maxJsonRawLength)
+			}
```
Comment (elgnay) [inline pkg/spoke/statusfeedback/reader.go]: It will be clearer if the error message includes the name of the JSON path.

**Comment by maleck13** [comment]
@qiujian16 great to see this change landing so quickly. Once it merges will give it a try. Is the work agent feature gate a start up flag?

**Comment by qiujian16** [comment]
> @qiujian16 great to see this change landing so quickly. Once it merges will give it a try. Is the work agent feature gate a start up flag?

yes, in the latest version of clusteradm, you can run `clustadm join ... --feature-gates=RawFeedbackJsonString=true` to enable it

**Pull request merged**